src/lib.rs

```diff
@@ -436,35 +566,48 @@ impl SimpleAccumulator {
                 for i in temp_values.iter().skip(start_fill_index) {
                     self.last_write_position = (self.last_write_position + 1) % self.capacity;
                     sum += self.vec[self.last_write_position];
-                    sq_sum +=
-                        self.vec[self.last_write_position] * self.vec[self.last_write_position];
+                    sq_sum += self.vec[self.last_write_position] * self.vec[self.last_write_position];
                     self.vec[self.last_write_position] = *i;
                 }
-                self.mean =
-                    ((self.mean * (self.len + temp_values.len()) as f64) - sum) / self.len as f64;
+                
+                self.mean = ((self.mean * (self.len + temp_values.len()) as f64) - sum) / self.len as f64;
 
                 self.population_variance = (self.len as f64
                     * (old_variance + old_old_mean * old_old_mean)
                     + (-1.0) * (sq_sum + self.len as f64 * self.mean * self.mean)
                     + old_sq_sum)
-                    / self.len as f64;
+                    / self.len as f64;         
             }
-        } else {
-            // normal append
+        } 
+        // Using vector append when fixed_capacity is 'false'
+        else {
             self.len += temp_values.len();
             self.vec.append(&mut temp_values);
             self.capacity = self.vec.capacity();
+            //self.calculate_approx_median();
+        }
+
+        self.stats[0] = self.mean;
+        self.stats[1] = self.population_variance * self.len as f64;
+
+        if self.accumulate {
```

Comment:
Does this iterate over whole vec and calculate for each push and pop?

Reply:
This code snippet occurs in the append function. In the append function, you have code that computes mean and variance. In my addition, I have a vector called stats that has 4 entries. These 4 entries help in online computation when a single entry is pushed or popped. In the append function, your code for mean and variance goes perfectly, and the online functions I added (functions with the _online suffix) don't need to be called. But for future push and pop operations, I need to keep the stats vector values updated. Line 590, 591 does exactly that.

Line 593 is just a tiny refactoring where I am calling calculate_approx_median on self when the accumulate flag is set, irrespective of the accumulator size being fixed or not, since it was being called individually inside multiple ifs, before. And I also call my offline functions calculate_skewness etc on the data of the accumulator. These functions not only keep the skewness etc fields updated, but also update the entries 2 and 3 of the stats vector, which, as I explained earlier, is important for future push and pops.  

Please let me know if the above explanation did not make sense. We can also meet and clear up issues about my PR, if you would prefer that.

Reply:
I mean `calculate_skewness` function called at 595, it iterates through whole vec right?

Reply:
Hello, I made the necessary changes in append() now (last moment, I know 🤦‍♀️ ). Check it whenever you have time 🥲

Reply:
![Screenshot 2023-05-19 164704](https://github.com/SubconsciousCompute/SimpleAccumulator/assets/4660526/6caf9912-7954-41d5-b6c0-cb6e91d204fc)
![Screenshot 2023-05-19 164753](https://github.com/SubconsciousCompute/SimpleAccumulator/assets/4660526/a244cee6-71a0-4e5c-8bfd-876166253a66)

Reply:
@purnatag Does this also work when capacity is fixed and we start rewriting in FIFO order?